android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboard.java

```diff
@@ -425,11 +425,12 @@ public boolean onTouchEvent(MotionEvent event) {
 
   public void onResume() {
     DisplayMetrics dms = context.getResources().getDisplayMetrics();
-    int kbWidth = (int) (dms.widthPixels / dms.density);
-    // Ensure window is loaded for javascript functions
-    loadJavascript(KMString.format(
-      "window.onload = function(){ setOskWidth(\"%d\");"+
-      "setOskHeight(\"0\"); };", kbWidth));
+    // Keyboard width = device width.
+    int kbWidth = (int) dms.widthPixels;
```

Comment:
I am not familiar with this code base, how come the kbWidth is no longer (dms.widthPixels/ dms.density)?

Reply:
I found it quite strange that we were doing the calculation for width differently than for height.  We should be consistent about that.

Furthermore, code was already in place to actually _do_ the calculation where needed in a manner consistent with height... so dividing by density here didn't even make sense.

Notice `setOskWidth` and `setOskHeight` within android-host.js - that's where the density adjustment is performed.